SERVER-17410: listCommands returns unordered list

upendrag opened the pull request:
JIRA link: https://jira.mongodb.org/browse/SERVER-17410

Fix summary:
Construct the result object for "listCommands" using sorted commands map constructed from existing commands object.

jrassi commented:
Thanks again for submitting your pull request!

Could you please rebase this against a recent commit from master and re-submit?  We very recently auto-formatted the entire src/mongo/ directory with clang-format, so you'll notice that there have been significant whitespace changes in the file being modified.  Thanks.

upendrag commented:
I did a rebase against the recent commit from master and had to merge it with my branch, since upendrag::server-17410 and mongo:: master were divided. I have addressed review comments too.

upendrag commented:
Since there are lot file changes due to the recent clang formatting that happened between my commits, I am closing this and will open a new one.
